nav2_smac_planner/include/nav2_smac_planner/smoother.hpp

```diff
@@ -72,8 +73,33 @@ class Smoother
     nav_msgs::msg::Path & path,
     const nav2_costmap_2d::Costmap2D * costmap,
     const double & max_time,
-    const bool do_refinement = true)
+    const bool do_refinement = true, const bool single_dir_segment = false)
   {
+    if (!single_dir_segment) {
```

Comment:
This all technically works... but I would avoid recursion if not required. This could just be made to put the core smoothing logic below into a separate `smoothSegment` function and then just loop over it with the various segments and attach them in the end, without need for recursion.  The `do_refinement` was necessary to have recursion in unfortunately, but this can be modeled more clearly without it.

Reply:
I thought about doing that but went for recursion to minimize the amount of changes. I can reorganize the logic once I find a bit of spare time